src/openeo_gfmap/manager/job_manager.py

```diff
@@ -150,24 +202,40 @@ def _resume_postjob_actions(self, df: pd.DataFrame):
             job = connection.job(row.id)
             if row.status == "postprocessing":
                 _log.info(
-                    f"Resuming postprocessing of job {row.id}, queueing on_job_finished..."
+                    "Resuming postprocessing of job %s, queueing on_job_finished...",
+                    row.id,
+                )
+                future = self._executor.submit(self.on_job_done, job, row, _stac_lock)
```

Comment:
out of curiosity, what does _stack_lock do here, that is not needed in case of an error?

Reply:
Because we use multi-threading to run the `on_job_done` (basically downloading data, running user-defined post_job_action and adding items in the stac catalog). All those tasks are separated except the stac catalogue. You don't want to have two threads writing in the stac catalog at the same time or you would have a race condition.